libsolidity/interface/CompilerStack.cpp

```diff
@@ -466,7 +469,12 @@ eth::AssemblyItems const* CompilerStack::assemblyItems(string const& _contractNa
 eth::AssemblyItems const* CompilerStack::runtimeAssemblyItems(string const& _contractName) const
 {
 	if (m_stackState != CompilationSuccessful)
-		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Compilation was not successful."));
+	{
+		if (m_parserErrorRecovery)
```

Comment:
We should never reach `CompilationSuccessful` if there are errors, so I think this change can be reverted.

Reply:
Reverted.